widget/charts/src/components/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,381 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+import type {
+  BarStackDataType,
+  PropTypes,
+  TooltipDataType,
+} from './BarChart.types';
+import type { BarGroupBar, SeriesPoint } from '@visx/shape/lib/types';
+
+import { Divider } from '@rango-dev/ui';
+import { AxisBottom, AxisLeft } from '@visx/axis';
+import { localPoint } from '@visx/event';
+import { Grid } from '@visx/grid';
+import { Group } from '@visx/group';
+import { scaleBand, scaleLinear, scaleOrdinal } from '@visx/scale';
+import { BarStack } from '@visx/shape';
+import { useTooltip, useTooltipInPortal } from '@visx/tooltip';
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import React, { Fragment, useEffect, useRef } from 'react';
+
+import {
+  AmountConverter,
+  compactNumberFormat,
+} from '../../utils/amountConverter';
+import { getDayOfMonth } from '../../utils/common';
+
+import {
+  DEFAULT_MARGIN,
+  MOBILE_BREAKPOINT,
+  NINETY_DAYS,
+  SEVEN_DAYS,
+  TOOLTIP_DELAY_MS,
+  TOOLTIP_HIDE_DELAY_MS,
+  TOOLTIP_SHADOW,
+} from './BarChart.constants';
+import {
+  bottomAxisLayout,
+  generateTickValues,
+  getTotalValue,
+  getTotalValueDates,
+} from './BarChart.helpers';
+import {
+  Circle,
+  Container,
+  InfoContainer,
+  Line,
+  NameWrapper,
+  TooltipContainer,
+  TooltipInfoRow,
+} from './BarChart.styles';
+
+dayjs.extend(utc);
+
+export const BarChart = (props: PropTypes) => {
+  const {
+    data,
+    type,
+    days,
+    width,
+    height,
+    margin = DEFAULT_MARGIN,
+    colorBlockchainMap,
+    buckets,
```

Comment:
The logic of `days` and `type` is not related to this component.
And the consumer should handle it.

It is also better to change the name of `colorBlockchainMap`,
for example, something like `colorBucketMap`.

Reply:
fixed